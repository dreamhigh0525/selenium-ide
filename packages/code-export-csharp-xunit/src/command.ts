// Licensed to the Software Freedom Conservancy (SFC) under one
// or more contributor license agreements.  See the NOTICE file
// distributed with this work for additional information
// regarding copyright ownership.  The SFC licenses this file
// to you under the Apache License, Version 2.0 (the
// "License"); you may not use this file except in compliance
// with the License.  You may obtain a copy of the License at
//
//   http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing,
// software distributed under the License is distributed on an
// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
// KIND, either express or implied.  See the License for the
// specific language governing permissions and limitations
// under the License.

import { Command, location } from '@seleniumhq/code-export-csharp-commons'
import { codeExport as exporter } from '@seleniumhq/side-code-export'
import { CommandShape } from '@seleniumhq/side-model'

const emitters = { ...Command.emitters }

exporter.register.preprocessors(emitters)

function emit(command: CommandShape) {
  return exporter.emit.command(command, emitters[command.command], {
    variableLookup: Command.variableLookup,
    emitNewWindowHandling: Command.extras.emitNewWindowHandling,
  })
}

emitters.assert = emitAssert
emitters.verify = emitAssert

function emitAssert(varName: string, value: string) {
  return Promise.resolve(
    `Assert.Equal(vars["${varName}"].ToString(), "${value}");`
  )
}

emitters.assertAlert = emitAssertAlert
emitters.assertConfirmation = emitAssertAlert
emitters.assertPrompt = emitAssertAlert

function emitAssertAlert(AlertText: string) {
  return Promise.resolve(
    `Assert.Equal(driver.SwitchTo().Alert().Text, "${AlertText}");`
  )
}

emitters.assertChecked = emitVerifyChecked
emitters.verifyChecked = emitVerifyChecked

async function emitVerifyChecked(locator: string) {
  return Promise.resolve(
    `Assert.True(driver.FindElement(${await location.emit(locator)}).Selected);`
  )
}

emitters.assertEditable = emitVerifyEditable
emitters.verifyEditable = emitVerifyEditable

async function emitVerifyEditable(locator: string) {
  const commands = [
    { level: 0, statement: '{' },
    {
      level: 1,
      statement: `var element = driver.FindElement(${await location.emit(
        locator
      )});`,
    },
    {
      level: 1,
      statement:
        'Boolean isEditable = element.Enabled && element.GetAttribute("readonly") == null;',
    },
    { level: 1, statement: 'Assert.True(isEditable);' },
    { level: 0, statement: '}' },
  ]
  return Promise.resolve({ commands })
}

emitters.assertElementPresent = emitVerifyElementPresent
emitters.verifyElementPresent = emitVerifyElementPresent

async function emitVerifyElementPresent(locator: string) {
  const commands = [
    { level: 0, statement: '{' },
    {
      level: 1,
      statement: `IReadOnlyCollection<IWebElement> elements = driver.FindElements(${await location.emit(
        locator
      )});`,
    },
    { level: 1, statement: 'Assert.True(elements.Count > 0);' },
    { level: 0, statement: '}' },
  ]
  return Promise.resolve({ commands })
}

emitters.assertElementNotPresent = emitVerifyElementNotPresent
emitters.verifyElementNotPresent = emitVerifyElementNotPresent

async function emitVerifyElementNotPresent(locator: string) {
  const commands = [
    { level: 0, statement: '{' },
    {
      level: 1,
      statement: `IReadOnlyCollection<IWebElement> elements = driver.FindElements(${await location.emit(
        locator
      )});`,
    },
    { level: 1, statement: 'Assert.True(elements.Count == 0);' },
    { level: 0, statement: '}' },
  ]
  return Promise.resolve({ commands })
}

emitters.assertNotChecked = emitVerifyNotChecked
emitters.verifyNotChecked = emitVerifyNotChecked

async function emitVerifyNotChecked(locator: string) {
  return Promise.resolve(
    `Assert.False(driver.FindElement(${await location.emit(
      locator
    )}).Selected);`
  )
}

emitters.assertNotEditable = emitVerifyNotEditable
emitters.verifyNotEdtiable = emitVerifyNotEditable

async function emitVerifyNotEditable(locator: string) {
  const commands = [
    { level: 0, statement: '{' },
    {
      level: 1,
      statement: `var element = driver.FindElement(${await location.emit(
        locator
      )});`,
    },
    {
      level: 1,
      statement:
        'Boolean isEditable = element.Enabled && element.GetAttribute("readonly") == null;',
    },
    { level: 1, statement: 'Assert.False(isEditable);' },
    { level: 0, statement: '}' },
  ]
  return Promise.resolve({ commands })
}

emitters.assertNotSelectedValue = emitVerifyNotSelectedValue
emitters.verifyNotSelectedValue = emitVerifyNotSelectedValue

async function emitVerifyNotSelectedValue(locator: string, expectedValue: string) {
  const commands = [
    { level: 0, statement: '{' },
    {
      level: 1,
      statement: `String value = driver.FindElement(${await location.emit(
        locator
      )}).GetAttribute("value");`,
    },
    {
      level: 1,
      statement: `Assert.NotEqual(value, ${exporter.emit.text(
        expectedValue
      )});`,
    },
    { level: 0, statement: '}' },
  ]
  return Promise.resolve({ commands })
}

emitters.assertNotText = emitVerifyNotText
emitters.verifyNotText = emitVerifyNotText

async function emitVerifyNotText(locator: string, text: string) {
  const result = `driver.FindElement(${await location.emit(locator)}).Text`
  return Promise.resolve(
    `Assert.NotEqual(${result}, "${exporter.emit.text(text)}");`
  )
}

emitters.assertSelectedLabel = emitVerifySelectedLabel
emitters.verifySelectedLabel = emitVerifySelectedLabel

async function emitVerifySelectedLabel(locator: string, labelValue: string) {
  const commands = [
    { level: 0, statement: '{' },
    {
      level: 1,
      statement: `var element = driver.FindElement(${await location.emit(
        locator
      )});`,
    },
    { level: 1, statement: 'String value = element.GetAttribute("value");' },
    {
      level: 1,
      statement: `String locator = String.Format("option[@value='%s']", "value");`,
    },
    {
      level: 1,
      statement:
        'String selectedText = element.FindElement(By.XPath(locator)).Text;',
    },
    {
      level: 1,
      statement: `Assert.Equal(selectedText, "${labelValue}");`,
    },
    { level: 0, statement: '}' },
  ]
  return Promise.resolve({
    commands,
  })
}

emitters.assertSelectedValue = emitVerifyValue
emitters.verifySelectedValue = emitVerifyValue
emitters.assertValue = emitVerifyValue

async function emitVerifyValue(locator: string, value: string) {
  const commands = [
    { level: 0, statement: '{' },
    {
      level: 1,
      statement: `String value = driver.FindElement(${await location.emit(
        locator
      )}).GetAttribute("value");`,
    },
    { level: 1, statement: `Assert.Equal(value, "${value}");` },
    { level: 0, statement: '}' },
  ]
  return Promise.resolve({ commands })
}

emitters.assertText = emitVerifyText
emitters.verifyText = emitVerifyText

async function emitVerifyText(locator: string, text: string) {
  return Promise.resolve(
    `Assert.Equal(driver.FindElement(${await location.emit(
      locator
    )}).Text, "${exporter.emit.text(text)}");`
  )
}

emitters.assertTitle = emitVerifyTitle
emitters.verifyTitle = emitVerifyTitle

async function emitVerifyTitle(title: string) {
  return Promise.resolve(`Assert.Equal(driver.Title, "${title}");`)
}

export default {
  ...Command,
  emit,
  emitters,
}
