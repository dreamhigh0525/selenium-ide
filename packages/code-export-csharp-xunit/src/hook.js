// Licensed to the Software Freedom Conservancy (SFC) under one
// or more contributor license agreements.  See the NOTICE file
// distributed with this work for additional information
// regarding copyright ownership.  The SFC licenses this file
// to you under the Apache License, Version 2.0 (the
// "License"); you may not use this file except in compliance
// with the License.  You may obtain a copy of the License at
//
//   http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing,
// software distributed under the License is distributed on an
// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
// KIND, either express or implied.  See the License for the
// specific language governing permissions and limitations
// under the License.

import { codeExport as exporter } from '@seleniumhq/side-utils'

const emitters = {
  afterAll,
  afterEach,
  beforeAll,
  beforeEach,
  declareDependencies,
  declareMethods: empty,
  declareVariables,
  inEachBegin: empty,
  inEachEnd: empty,
}

function generate(hookName) {
  return new exporter.hook(emitters[hookName]())
}

export function generateHooks() {
  let result = {}
  Object.keys(emitters).forEach(hookName => {
    result[hookName] = generate(hookName)
  })
  return result
}

function afterAll() {
  const params = {
    startingSyntax: {
      commands: [{ level: 0, statement: 'public void finalTearDown() {' }],
    },
    endingSyntax: {
      commands: [{ level: 0, statement: '}' }],
    },
    registrationLevel: 1,
  }
  return params
}

function afterEach() {
  const params = {
    startingSyntax: {
      commands: [
        { level: 0, statement: 'public void tearDown() {' },
        { level: 1, statement: 'driver.Quit();' },
        { level: 1, statement: 'driver.Dispose();' },
      ],
    },
    endingSyntax: {
      commands: [{ level: 0, statement: '}' }],
    },
  }
  return params
}

function beforeAll() {
  const params = {
    startingSyntax: {
      commands: [
        { level: 0, statement: '[Fact]' }
        { level: 0, statement: 'public void initialSetUp() {' }
      ],
    },
    endingSyntax: {
      commands: [{ level: 0, statement: '}' }],
    },
    registrationLevel: 1,
  }
  return params
}

function beforeEach() {
  const params = {
    startingSyntax: {
      commands: [
        { level: 0, statement: 'public void setUp() {' },
        { level: 1, statement: 'driver = new FirefoxDriver();' },
        { level: 1, statement: 'js = (IJavascriptExecutor) driver;' },
        { level: 1, statement: 'vars = new Dictionary<String, Object>();' },
      ],
    },
    endingSyntax: {
      commands: [{ level: 0, statement: '}' }],
    },
  }
  return params
}

function declareDependencies() {
  const params = {
    startingSyntax: {
      commands: [
        { level: 0, statement: 'using System;' },
        { level: 0, statement: 'using System.Collections.Generic;' },
        { level: 0, statement: 'using OpenQA.Selenium;' },
        { level: 0, statement: 'using OpenQA.Selenium.Chrome;' },
        { level: 0, statement: 'using OpenQA.Selenium.Firefox;' },
        { level: 0, statement: 'using OpenQA.Selenium.Support.UI;' },
        { level: 0, statement: 'using OpenQA.Selenium.Interactions;' },
        { level: 0, statement: 'using Xunit;' },
      ],
    },
  }
  return params
}

function declareVariables() {
  const params = {
    startingSyntax: {
      commands: [
        { level: 0, statement: 'private IWebDriver driver;' },
        {
          level: 0,
          statement: 'private IDictionary<String, Object> vars;',
        },
        {
          level: 0,
          statement: 'IJavaScriptExecutor js;',
        },
      ],
    },
  }
  return params
}

function empty() {
  return {}
}
