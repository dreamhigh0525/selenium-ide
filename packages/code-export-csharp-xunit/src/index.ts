// Licensed to the Software Freedom Conservancy (SFC) under one
// or more contributor license agreements.  See the NOTICE file
// distributed with this work for additional information
// regarding copyright ownership.  The SFC licenses this file
// to you under the Apache License, Version 2.0 (the
// "License"); you may not use this file except in compliance
// with the License.  You may obtain a copy of the License at
//
//   http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing,
// software distributed under the License is distributed on an
// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
// KIND, either express or implied.  See the License for the
// specific language governing permissions and limitations
// under the License.

import {
  codeExport as exporter,
  generateHooks,
  LanguageEmitter,
  LanguageEmitterOpts,
} from '@seleniumhq/side-code-export'
import emitter from './command'
import hooks from './hook'
import { location } from '@seleniumhq/code-export-csharp-commons'

// Define language options
export const displayName = 'C# xUnit'

export const opts: LanguageEmitterOpts = {
  emitter: emitter,
  displayName,
  name: 'csharp-xunit',
  hooks: generateHooks(hooks),
  fileExtension: '.cs',
  commandPrefixPadding: '  ',
  terminatingKeyword: '}',
  commentPrefix: '//',
  generateFilename: (name) => {
    return `${exporter.parsers.capitalize(
      exporter.parsers.sanitizeName(name)
    )}Test${opts.fileExtension}`
  },

  generateMethodDeclaration: (name) => {
    return `public void ${exporter.parsers.sanitizeName(name)}() \n{`
  },
  generateTestDeclaration: (name) => {
    return `[Fact]\npublic void ${exporter.parsers.capitalize(
      exporter.parsers.sanitizeName(name)
    )}() {`
  },
  generateSuiteDeclaration: () => {
    return `public class SuiteTests : IDisposable {`
  },
}

const language: LanguageEmitter = {
  emit: {
    test: async function emitTest({
      test,
      tests,
      project,
      enableOriginTracing,
      beforeEachOptions,
      enableDescriptionAsComment,
    }) {
      const testDeclaration = opts.generateTestDeclaration(test.name)
      const result = await exporter.emit.test(test, tests, {
        ...opts,
        testDeclaration,
        enableOriginTracing,
        enableDescriptionAsComment,
        project,
      })
      const suiteName = test.name
      const suiteDeclaration = opts.generateSuiteDeclaration(suiteName)
      var _suite = await exporter.emit.suite(result, tests, {
        ...opts,
        suiteDeclaration,
        suiteName,
        project,
        beforeEachOptions,
      })
      return {
        filename: opts.generateFilename(test.name),
        body: exporter.emit.orderedSuite(_suite),
      }
    },
    suite: async function emitSuite({
      suite,
      tests,
      project,
      enableOriginTracing,
      beforeEachOptions,
      enableDescriptionAsComment,
    }) {
      const result = await exporter.emit.testsFromSuite(tests, suite, opts, {
        enableOriginTracing,
        enableDescriptionAsComment,
        generateTestDeclaration: opts.generateTestDeclaration,
        project,
      })
      const suiteDeclaration = opts.generateSuiteDeclaration(suite.name)
      var _suite = await exporter.emit.suite(result, tests, {
        ...opts,
        suiteDeclaration,
        suite,
        suiteName: suite.name,
        project,
        beforeEachOptions,
      })
      return {
        filename: opts.generateFilename(suite.name),
        body: emitOrderedSuite(_suite, suite.name),
      }
    },
    locator: location.emit,
    command: emitter.emit,
  },
  opts,
  register: {
    command: emitter.register,
    variable: opts.hooks.declareVariables.register,
    dependency: opts.hooks.declareDependencies.register,
    beforeAll: opts.hooks.beforeAll.register,
    beforeEach: opts.hooks.beforeEach.register,
    afterEach: opts.hooks.afterEach.register,
    afterAll: opts.hooks.afterAll.register,
    inEachBegin: opts.hooks.inEachBegin.register,
    inEachEnd: opts.hooks.inEachEnd.register,
  },
}

function generateNamespace(name: string) {
  return `namespace ${exporter.parsers.capitalize(
    exporter.parsers.sanitizeName(name)
  )}\n{\n`
}

function emitOrderedSuite(emittedSuite: any, name: string) {
  let result = ''
  result += emittedSuite.headerComment
  result += emittedSuite.dependencies
  result += `${generateNamespace(name)}`
  result += emittedSuite.suiteDeclaration
  result += emittedSuite.variables
  //result += emittedSuite.beforeAll
  result += emittedSuite.beforeEach
  result += emittedSuite.afterEach
  //result += emittedSuite.afterAll
  result += emittedSuite.methods
  if (emittedSuite.tests.testDeclaration) {
    const test = emittedSuite.tests
    result += test.testDeclaration
    result += test.inEachBegin
    result += test.commands
    result += test.inEachEnd
    result += test.testEnd
  } else {
    for (const testName in emittedSuite.tests) {
      const test = emittedSuite.tests[testName]
      result += test.testDeclaration
      result += test.inEachBegin
      result += test.commands
      result += test.inEachEnd
      result += test.testEnd
    }
  }
  result += emittedSuite.suiteEnd
  result += '}' // terminate namespace
  return result
}

export default language
