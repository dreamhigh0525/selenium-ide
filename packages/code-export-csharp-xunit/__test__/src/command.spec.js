// Licensed to the Software Freedom Conservancy (SFC) under one
// or more contributor license agreements.  See the NOTICE file
// distributed with this work for additional information
// regarding copyright ownership.  The SFC licenses this file
// to you under the Apache License, Version 2.0 (the
// "License"); you may not use this file except in compliance
// with the License.  You may obtain a copy of the License at
//
//   http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing,
// software distributed under the License is distributed on an
// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
// KIND, either express or implied.  See the License for the
// specific language governing permissions and limitations
// under the License.

import Command from '../../src/command'
import { Commands } from '../../../selenium-ide/src/neo/models/Command'
import { opts } from '../../src/index'
import { codeExport as exporter } from '@seleniumhq/side-code-export'

const commandPrefixPadding = opts.commandPrefixPadding

async function prettify(command, { fullPayload } = {}) {
  const commandBlock = await Command.emit(command)
  const result = exporter.prettify(commandBlock, {
    commandPrefixPadding,
  })
  return fullPayload ? result : result.body
}

describe('command code emitter', () => {
  it.skip('should emit all known commands', () => {
    let result = []
    Commands.array.forEach(command => {
      if (!Command.canEmit(command)) {
        result.push(command)
      }
    })
    expect(() => {
      if (result.length) {
        if (result.length === 1) {
          throw new Error(`${result[0]} has no emitter, write one!`)
        } else {
          throw new Error(`No emitters for ${result.join(', ')}. Write them!`)
        }
      }
    }).not.toThrow()
  })
  it('should emit `assert` command', () => {
    const command = {
      command: 'assert',
      target: 'varrrName',
      value: 'blah',
    }
    return expect(prettify(command)).resolves.toMatchSnapshot()
  })
  it('should emit `assert alert` command', () => {
    const command = {
      command: 'assertAlert',
      target: 'an alert',
      value: '',
    }
    return expect(prettify(command)).resolves.toMatchSnapshot()
  })
  it('should emit `assert checked` command', () => {
    const command = {
      command: 'assertChecked',
      target: 'id=check',
      value: '',
    }
    return expect(prettify(command)).resolves.toMatchSnapshot()
  })
  it('should emit `assert confirmation` command', () => {
    const command = {
      command: 'assertConfirmation',
      target: 'a confirmation',
      value: '',
    }
    return expect(prettify(command)).resolves.toMatchSnapshot()
  })
  it('should emit `assert editable` command', () => {
    const command = {
      command: 'assertEditable',
      target: 'id=text',
      value: '',
    }
    return expect(prettify(command)).resolves.toMatchSnapshot()
  })
  it('should emit `assert element present` command', () => {
    const command = {
      command: 'assertElementPresent',
      target: 'id=element',
      value: '',
    }
    return expect(prettify(command)).resolves.toMatchSnapshot()
  })
  it('should emit `assert element not present` command', () => {
    const command = {
      command: 'assertElementNotPresent',
      target: 'id=element',
      value: '',
    }
    return expect(prettify(command)).resolves.toMatchSnapshot()
  })
  it('should emit `assert not checked` command', () => {
    const command = {
      command: 'assertNotChecked',
      target: 'id=check',
      value: '',
    }
    return expect(prettify(command)).resolves.toMatchSnapshot()
  })
  it('should emit `assert not editable` command', () => {
    const command = {
      command: 'assertNotEditable',
      target: 'id=text',
      value: '',
    }
    return expect(prettify(command)).resolves.toMatchSnapshot()
  })
  it('should emit `assert not selected value` command', () => {
    const command = {
      command: 'assertNotSelectedValue',
      target: 'id=select',
      value: 'test',
    }
    return expect(prettify(command)).resolves.toMatchSnapshot()
  })
  it('should emit `assert not text` command', () => {
    const command = {
      command: 'assertNotText',
      target: 'id=test',
      value: 'text',
    }
    return expect(prettify(command)).resolves.toMatchSnapshot()
  })
  it('should emit `assert prompt` command', () => {
    const command = {
      command: 'assertPrompt',
      target: 'a prompt',
      value: '',
    }
    return expect(prettify(command)).resolves.toMatchSnapshot()
  })
  it("should emit 'assert selected label' command", () => {
    const command = {
      command: 'assertSelectedLabel',
      target: 'id=test',
      value: 'test',
    }
    return expect(prettify(command)).resolves.toMatchSnapshot()
  })
  it('should emit `assert selected value` command', () => {
    const command = {
      command: 'assertSelectedValue',
      target: 'id=select',
      value: 'test',
    }
    return expect(prettify(command)).resolves.toMatchSnapshot()
  })
  it('should emit `assert text` command', () => {
    const command = {
      command: 'assertText',
      target: 'id=test',
      value: 'some text that should be here',
    }
    return expect(prettify(command)).resolves.toMatchSnapshot()
  })
  it('should emit `assert title` command', () => {
    const command = {
      command: 'assertTitle',
      target: 'example title',
      value: '',
    }
    return expect(prettify(command)).resolves.toMatchSnapshot()
  })
  it('should emit `assert value` command', () => {
    const command = {
      command: 'assertValue',
      target: 'id=select',
      value: 'test',
    }
    return expect(prettify(command)).resolves.toMatchSnapshot()
  })
  it('should emit `verify` command', () => {
    const command = {
      command: 'verify',
      target: 'varrrName',
      value: 'blah',
    }
    return expect(prettify(command)).resolves.toMatchSnapshot()
  })
  it('should emit `verify checked` command', () => {
    const command = {
      command: 'verifyChecked',
      target: 'id=check',
      value: '',
    }
    return expect(prettify(command)).resolves.toMatchSnapshot()
  })
  it('should emit `verify editable` command', () => {
    const command = {
      command: 'verifyEditable',
      target: 'id=text',
      value: '',
    }
    return expect(prettify(command)).resolves.toMatchSnapshot()
  })
  it('should emit `verify element present` command', () => {
    const command = {
      command: 'verifyElementPresent',
      target: 'id=element',
      value: '',
    }
    return expect(prettify(command)).resolves.toMatchSnapshot()
  })
  it('should emit `verify element not present` command', () => {
    const command = {
      command: 'verifyElementNotPresent',
      target: 'id=element',
      value: '',
    }
    return expect(prettify(command)).resolves.toMatchSnapshot()
  })
  it('should emit `verify not checked` command', () => {
    const command = {
      command: 'verifyNotChecked',
      target: 'id=check',
      value: '',
    }
    return expect(prettify(command)).resolves.toMatchSnapshot()
  })
  it('should emit `verify not editable` command', () => {
    const command = {
      command: 'verifyNotEditable',
      target: 'id=text',
      value: '',
    }
    return expect(prettify(command)).resolves.toMatchSnapshot()
  })
  it('should emit `verify not selected value` command', () => {
    const command = {
      command: 'verifyNotSelectedValue',
      target: 'id=select',
      value: 'test',
    }
    return expect(prettify(command)).resolves.toMatchSnapshot()
  })
  it('should emit `verify not text` command', () => {
    const command = {
      command: 'verifyNotText',
      target: 'id=test',
      value: 'text',
    }
    return expect(prettify(command)).resolves.toMatchSnapshot()
  })
  it("should emit 'verify selected label' command", () => {
    const command = {
      command: 'verifySelectedLabel',
      target: 'id=test',
      value: 'test',
    }
    return expect(prettify(command)).resolves.toMatchSnapshot()
  })
  it('should emit `verify value` command', () => {
    const command = {
      command: 'verifyValue',
      target: 'id=select',
      value: 'test',
    }
    return expect(prettify(command)).resolves.toMatchSnapshot()
  })
  it('should emit `verify selected value` command', () => {
    const command = {
      command: 'verifySelectedValue',
      target: 'id=select',
      value: 'test',
    }
    return expect(prettify(command)).resolves.toMatchSnapshot()
  })
  it('should emit `verify text` command', () => {
    const command = {
      command: 'verifyText',
      target: 'id=test',
      value: 'some text that should be here',
    }
    return expect(prettify(command)).resolves.toMatchSnapshot()
  })
  it('should emit `verify title` command', () => {
    const command = {
      command: 'verifyTitle',
      target: 'example title',
      value: '',
    }
    return expect(prettify(command)).resolves.toMatchSnapshot()
  })
})
