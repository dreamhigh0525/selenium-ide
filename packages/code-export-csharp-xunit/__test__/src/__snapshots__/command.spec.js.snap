// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`command code emitter should emit 'assert selected label' command 1`] = `
"{
  var element = driver.FindElement(By.Id(\\"test\\"));
  String value = element.GetAttribute(\\"value\\");
  String locator = String.Format(\\"option[@value='%s']\\", \\"value\\");
  String selectedText = element.FindElement(By.XPath(locator)).Text;
  Assert.Equal(selectedText, \\"test\\");
}"
`;

exports[`command code emitter should emit 'verify selected label' command 1`] = `
"{
  var element = driver.FindElement(By.Id(\\"test\\"));
  String value = element.GetAttribute(\\"value\\");
  String locator = String.Format(\\"option[@value='%s']\\", \\"value\\");
  String selectedText = element.FindElement(By.XPath(locator)).Text;
  Assert.Equal(selectedText, \\"test\\");
}"
`;

exports[`command code emitter should emit \`assert alert\` command 1`] = `"Assert.Equal(driver.SwitchTo().Alert().Text, \\"an alert\\");"`;

exports[`command code emitter should emit \`assert checked\` command 1`] = `"Assert.True(driver.FindElement(By.Id(\\"check\\")).Selected);"`;

exports[`command code emitter should emit \`assert confirmation\` command 1`] = `"Assert.Equal(driver.SwitchTo().Alert().Text, \\"a confirmation\\");"`;

exports[`command code emitter should emit \`assert editable\` command 1`] = `
"{
  var element = driver.FindElement(By.Id(\\"text\\"));
  Boolean isEditable = element.Enabled && element.GetAttribute(\\"readonly\\") == null;
  Assert.True(isEditable);
}"
`;

exports[`command code emitter should emit \`assert element not present\` command 1`] = `
"{
  IReadOnlyCollection<IWebElement> elements = driver.FindElements(By.Id(\\"element\\"));
  Assert.True(elements.Count == 0);
}"
`;

exports[`command code emitter should emit \`assert element present\` command 1`] = `
"{
  IReadOnlyCollection<IWebElement> elements = driver.FindElements(By.Id(\\"element\\"));
  Assert.True(elements.Count > 0);
}"
`;

exports[`command code emitter should emit \`assert not checked\` command 1`] = `"Assert.False(driver.FindElement(By.Id(\\"check\\")).Selected);"`;

exports[`command code emitter should emit \`assert not editable\` command 1`] = `
"{
  var element = driver.FindElement(By.Id(\\"text\\"));
  Boolean isEditable = element.Enabled && element.GetAttribute(\\"readonly\\") == null;
  Assert.False(isEditable);
}"
`;

exports[`command code emitter should emit \`assert not selected value\` command 1`] = `
"{
  String value = driver.FindElement(By.Id(\\"select\\")).GetAttribute(\\"value\\");
  Assert.NotEqual(value, test);
}"
`;

exports[`command code emitter should emit \`assert not text\` command 1`] = `"Assert.NotEqual(driver.FindElement(By.Id(\\"test\\")).Text, \\"text\\");"`;

exports[`command code emitter should emit \`assert prompt\` command 1`] = `"Assert.Equal(driver.SwitchTo().Alert().Text, \\"a prompt\\");"`;

exports[`command code emitter should emit \`assert selected value\` command 1`] = `
"{
  String value = driver.FindElement(By.Id(\\"select\\")).GetAttribute(\\"value\\");
  Assert.Equal(value, \\"test\\");
}"
`;

exports[`command code emitter should emit \`assert text\` command 1`] = `"Assert.Equal(driver.FindElement(By.Id(\\"test\\")).Text, \\"some text that should be here\\");"`;

exports[`command code emitter should emit \`assert title\` command 1`] = `"Assert.Equal(driver.Title, \\"example title\\");"`;

exports[`command code emitter should emit \`assert value\` command 1`] = `
"{
  String value = driver.FindElement(By.Id(\\"select\\")).GetAttribute(\\"value\\");
  Assert.Equal(value, \\"test\\");
}"
`;

exports[`command code emitter should emit \`assert\` command 1`] = `"Assert.Equal(vars[\\"varrrName\\"].ToString(), \\"blah\\");"`;

exports[`command code emitter should emit \`verify checked\` command 1`] = `"Assert.True(driver.FindElement(By.Id(\\"check\\")).Selected);"`;

exports[`command code emitter should emit \`verify editable\` command 1`] = `
"{
  var element = driver.FindElement(By.Id(\\"text\\"));
  Boolean isEditable = element.Enabled && element.GetAttribute(\\"readonly\\") == null;
  Assert.True(isEditable);
}"
`;

exports[`command code emitter should emit \`verify element not present\` command 1`] = `
"{
  IReadOnlyCollection<IWebElement> elements = driver.FindElements(By.Id(\\"element\\"));
  Assert.True(elements.Count == 0);
}"
`;

exports[`command code emitter should emit \`verify element present\` command 1`] = `
"{
  IReadOnlyCollection<IWebElement> elements = driver.FindElements(By.Id(\\"element\\"));
  Assert.True(elements.Count > 0);
}"
`;

exports[`command code emitter should emit \`verify not checked\` command 1`] = `"Assert.False(driver.FindElement(By.Id(\\"check\\")).Selected);"`;

exports[`command code emitter should emit \`verify not editable\` command 1`] = `
"{
  var element = driver.FindElement(By.Id(\\"text\\"));
  Boolean isEditable = element.Enabled && element.GetAttribute(\\"readonly\\") == null;
  Assert.False(isEditable);
}"
`;

exports[`command code emitter should emit \`verify not selected value\` command 1`] = `
"{
  String value = driver.FindElement(By.Id(\\"select\\")).GetAttribute(\\"value\\");
  Assert.NotEqual(value, test);
}"
`;

exports[`command code emitter should emit \`verify not text\` command 1`] = `"Assert.NotEqual(driver.FindElement(By.Id(\\"test\\")).Text, \\"text\\");"`;

exports[`command code emitter should emit \`verify selected value\` command 1`] = `
"{
  String value = driver.FindElement(By.Id(\\"select\\")).GetAttribute(\\"value\\");
  Assert.Equal(value, \\"test\\");
}"
`;

exports[`command code emitter should emit \`verify text\` command 1`] = `"Assert.Equal(driver.FindElement(By.Id(\\"test\\")).Text, \\"some text that should be here\\");"`;

exports[`command code emitter should emit \`verify title\` command 1`] = `"Assert.Equal(driver.Title, \\"example title\\");"`;

exports[`command code emitter should emit \`verify value\` command 1`] = `
"{
  string value = driver.FindElement(By.Id(\\"select\\")).GetAttribute(\\"value\\");
  Assert.That(value, Is.EqualTo(\\"test\\"));
}"
`;

exports[`command code emitter should emit \`verify\` command 1`] = `"Assert.Equal(vars[\\"varrrName\\"].ToString(), \\"blah\\");"`;
